File: src/main/java/org/minturtle/careersupport/common/utils/Base64Utils.java
package org.minturtle.careersupport.common.utils;

import org.springframework.stereotype.Component;

import java.nio.charset.StandardCharsets;
import java.util.Base64;

@Component
public class Base64Utils {


    public String encode(String str) {
        return Base64.getEncoder().encodeToString(str.getBytes(StandardCharsets.UTF_8));
    }

    public String decode(String str) {
        byte[] decodedBytes = Base64.getDecoder().decode(str);
        return new String(decodedBytes, StandardCharsets.UTF_8);
    }
}

File: src/main/kotlin/org/minturtle/careersupport/common/utils/Base64Utils.kt
package org.minturtle.careersupport.common.utils

import java.nio.charset.StandardCharsets
import java.util.*

object Base64Utils {

    fun encode(str: String): String {
        return Base64.getEncoder().encodeToString(str.toByteArray(StandardCharsets.UTF_8))
    }

    fun decode(str: String?): String {
        val decodedBytes = Base64.getDecoder().decode(str)
        return String(decodedBytes, StandardCharsets.UTF_8)
    }
}